Userland/Services/SpiceAgent/SpiceAgent.cpp

```diff
@@ -40,6 +41,59 @@ ErrorOr<void> SpiceAgent::start()
     auto capabilities_message = AnnounceCapabilitiesMessage(false, m_capabilities);
     TRY(this->send_message(capabilities_message));
 
+    GUI::Clipboard::the().on_change = [this](auto const& mime_type) {
+        auto result = this->on_clipboard_update(String::from_deprecated_string(mime_type).release_value_but_fixme_should_propagate_errors());
```

Comment:
🤔 this could be "resolved" with some trickery like

```
Optional<Error> err = {};
auto mime_str = String::from_deprecated_string(mime_type);
if (mime_str.is_error()) {
   err = mime_str.release_error();
} else {
    auto res = this->on_clipboard_update(mime_str.release_value());
    if (res.is_error()) {
        err = res.release_error();
    }
}
if (err.has_value())
   dbgln("boo, error {}", err);
```

or...

```
auto result = [&]() -> ErrorOr<void> {
   TRY(this->on_clipboard_update(TRY(String::from_deprecated_string(mime_type)));
}();
if (result.is_error()) {
    dbgln("Failed to inform the spice server of a clipboard update: {}", result.release_error());
}
```

.. but maybe it's best left for when we update the Clipboard to give you a String 😅

Reply:
I think we should wait until `GUI::Clipboard` provides a `String`, the conversion should be OK 99.9% of the time, except in scenarios where there is high memory usage and string allocation fails, but we probably have other issues by that point as well.